src/lib/core/overlay/overlay-state.ts

```diff
@@ -15,6 +15,9 @@ export class OverlayState {
   /** Strategy to be used when handling scroll events while the overlay is open. */
   scrollStrategy: ScrollStrategy = new NoopScrollStrategy();
 
+  /** Custom class to add to the dialog */
+  dialogClass: string = '';
```

Comment:
Overlay is the basis for multiple components, not just dialog. I would call this `panelClass`.